tests/aws/services/secretsmanager/test_secretsmanager.py

```diff
@@ -533,85 +628,91 @@ def test_rotate_secret_with_lambda_success(
-
-        self._wait_rotation(aws_client.secretsmanager, secret_name, rot_res["VersionId"])
-
-        response = aws_client.secretsmanager.describe_secret(SecretId=secret_name)
-        sm_snapshot.match("describe_secret_rotated", response)
-
-        list_secret_versions_1 = aws_client.secretsmanager.list_secret_version_ids(
-            SecretId=secret_name
+        self._assert_after_rotate_secret_with_lambda_success(
+            sm_snapshot,
+            secret_name,
+            secret_initial_version,
+            aws_client,
+            function_arn=function_arn,
+            rotation_config=rotation_config,
+        )
+        # Fetch the current version after first rotation
+        secret = aws_client.secretsmanager.get_secret_value(SecretId=secret_name)
+        self._assert_after_rotate_secret_with_lambda_success(
+            sm_snapshot,
+            secret_name,
+            secret["VersionId"],
+            aws_client,
+            snapshot_key_suffix="2",
+            function_arn=function_arn if set_rotation_on_second_run else None,
+            rotation_config=rotation_config if set_rotation_on_second_run else None,
         )
 
-        sm_snapshot.match("list_secret_versions_rotated_1", list_secret_versions_1)
-
     @markers.snapshot.skip_snapshot_verify(paths=["$..Error", "$..Message"])
     @markers.aws.validated
     def test_rotate_secret_invalid_lambda_arn(
         self, secret_name, aws_client, account_id, sm_snapshot
     ):
-        create_secret = aws_client.secretsmanager.create_secret(
-            Name=secret_name, SecretString="init"
-        )
-        sm_snapshot.add_transformer(
-            sm_snapshot.transform.secretsmanager_secret_id_arn(create_secret, 0)
-        )
-        sm_snapshot.match("create_secret", create_secret)
-
         region_name = aws_client.secretsmanager.meta.region_name
         invalid_arn = (
             f"arn:aws:lambda:{region_name}:{account_id}:function:rotate_secret_invalid_lambda_arn"
         )
-        with pytest.raises(Exception) as e:
-            aws_client.secretsmanager.rotate_secret(
-                SecretId=secret_name,
-                RotationLambdaARN=invalid_arn,
-                RotationRules={
-                    "AutomaticallyAfterDays": 1,
-                },
-            )
-        sm_snapshot.match("rotate_secret_invalid_arn_exc", e.value.response)
+        self._setup_invalid_rotation_secret(
+            invalid_arn, "rotate_secret_invalid_arn_exc", sm_snapshot, secret_name, aws_client
+        )
+        describe_secret = aws_client.secretsmanager.describe_secret(SecretId=secret_name)
+        sm_snapshot.match("describe_secret", describe_secret)
+        assert "RotationEnabled" not in describe_secret
+        assert "RotationRules" not in describe_secret
+        assert "RotationLambdaARN" not in describe_secret
 
+    @markers.snapshot.skip_snapshot_verify(paths=["$..Error", "$..Message"])
```

Comment:
issue: you have skipped the key part of this test => the error message. Can you remove this verify and implement the correct behaviour?

Reply:
Sure thing, I will remove the check it. The only thing that need to be updated is to add a check that raise an error `InvalidRequestException` that match AWS one